src/pubsub.c

```diff
@@ -615,6 +615,71 @@ void publishCommand(client *c) {
     addReplyLongLong(c,receivers);
 }
 
+void pubsubReplySubscribersClientInfo(client *c, list *clients) {
+    listNode *ln;
+    listIter li;
+    listRewind(clients, &li);
+    while ((ln = listNext(&li)) != NULL) {
+        client *client = listNodeValue(ln);
+        if (client->name) {
+            addReplyMapLen(c, 2);
+            addReplyBulkCString(c, "name");
+            addReplyBulk(c, client->name);
+        } else {
+            addReplyMapLen(c, 1);
+        }
+        addReplyBulkCString(c, "id");
+        addReplyBulkLongLong(c, client->id);
+    }
+}
+
+void pubsubReplySubscribers(client *c) {
+    sds type = c->argv[2]->ptr;
+    dict *d = NULL;
+    if (!strcmp(type, "global")) {
+        d = server.pubsub_channels;
+    } else if (!strcmp(type, "pattern")) {
+        d = server.pubsub_patterns;
+    } else if (!strcmp(type, "shard")) {
```

Comment:
We have a separate command for pattern subscription (PSUBSCRIBE vs SUBSCRIBE), should we be consistent and add a separate command (PSUBSCRIBERS vs SUBSCRIBERS) instead of adding nested parameters?

Reply:
Initially I thought about the same. With this option, we don't need to add subcommand for each subscription type (more boilerplate code). Also, by having an option in the API we could also introduce more options in the future if reqd.

Would like to stick with this option unless folks have a strong pushback.

Reply:
I'm okay either way, just wanted to call out that it's slightly inconsistent with existing commands.